Support 3.4 and key/value list

hanachin opened the pull request:
チケット: https://insidemine.agileware.jp/issues/19085

maimai77 commented on app/controllers/importer_controller.rb:
```diff
@@ -399,13 +399,21 @@ def result
 
+      ["start_date", "due_date"].each do |date_attr|
+        begin
+          issue.send("#{date_attr}=", Date.parse(row[attrs_map[date_attr]])) if row[attrs_map[date_attr]].present?
```
```ruby
issue.send("#{date_attr}=", row[attrs_map[date_attr]].try(:to_date))
```

eachで回さずにメソッドにした方が良いです。

mozamimy commented:
start_date と due_date の処理を分けるようにしました (114da8e)

mozamimy commented:
すみません、コメントしっかり読んでませんでした..
気をつけます。

```ruby
issue.send("#{date_attr}=", row[attrs_map[date_attr]].try(:to_date))
```

↑のロジックを `set_date_safely` メソッドに切り出し、`rescue 〜 end` 間で、attribute の名前だけが違う似たような処理が連続するので、`begin 〜 rescue 〜 end` 間をゴソッとメソッドとして切り出してみました。

maimai77 commented on app/controllers/importer_controller.rb:
```diff
@@ -563,4 +564,13 @@ def flash_message(type, text)
+      @failed_issues[@failed_count] = row
+      @messages << "Warning: When setting the #{date_attr} for issue #{@failed_count}, #{value} is not valid date"
+    end
+  end
```
良いと思います。
次に大きな修正が入ることがあれば、この辺りmodelに移した方がいいですね…

hanachin committed:
Fix NoMethodError: undefined method `parent=' for #<User:0x00007ffafb7434b8>

hanachin committed:
Fix ActiveRecord::RecordInvalid: Validation failed: Identifier cannot be blank, Identifier is too short (minimum is 1 characters)

hanachin committed:
Add .circleci/config.yml

hanachin committed:
Support key value list

hanachin committed:
.presence is not needed, fix test instead

hanachin committed:
Load Journal class before monkey patching

hanachin committed:
Remove unnecessary code

hanachin commented on app/controllers/importer_controller.rb:
```diff
@@ -731,13 +734,33 @@ def version_id_for_name!(project, name, add_versions)
+      if enumeration.nil?
+        if name.try! { |n| n.size > 1 } && add_enumerations
+          max_position = custom_field.enumerations.pluck(:position).max || 0
+          enumeration = CustomFieldEnumeration.create(custom_field: custom_field, name: name, position: max_position + 1).id
```
TODO: remove position as well as 4.0

hanachin committed:
Update orb

hanachin committed:
Remove add_enumeration checkbox

# Conflicts:
#	test/functional/importer_controller_test.rb

hanachin committed:
Do not send issue added notification when Send email notifications checkbox is unchecked

hanachin committed:
If user is not found, set nil to issue.assigned_to

hanachin committed:
If user is not found, set nil to user type custom field value

hanachin committed:
Update test

fkdkent commented:
Oops! There is a failed test...

hanachin committed:
Fix test

hanachin committed:
Clear Issue#assignet_to if is not assignable

fkdkent commented on test/functional/importer_controller_test.rb:
```diff
@@ -20,64 +20,164 @@ def setup
+    assert_equal @user, @issue.assigned_to
+  end
+
+  test 'should error when assigned_to user is not assignable' do
```
🔗 https://github.com/agileware-jp/redmine_importer/pull/5#discussion_r316026246

hanachin replied:
That is correct, thank you 🙏

hanachin committed:
Update test/functional/importer_controller_test.rb

Co-Authored-By: fkdkent <fkdkent@agileware.jp>

hanachin committed:
ユーザーが存在するがユーザー型のカスタムフィールドの値として設定できないとき、エラーにせず空欄にする

Fix https://agileware.slack.com/archives/CF7CGT7FH/p1566441841001200?thread_ts=1566370267.014800&cid=CF7CGT7FH

hanachin committed:
Reset primary key sequence in end of import to fix PG::UniqueViolation: ERROR

maimai77 committed:
Update app/controllers/importer_controller.rb

kyamada23 commented:
test ok

hanachin committed:
ガントチャートのチケット編集フォームなどからチケットを編集したとき、チケットに変更履歴が残るようにした

`Journal#empty?` メソッドが定義されていると `issue.current_journal.blank?` が違う値を返してしまうためうまく動かなくなる
https://github.com/rails/rails/blob/57ae917e9d52c9472fd37adbfa47d6de93dd01bf/activesupport/lib/active_support/core_ext/object/blank.rb#L19

agilemasayuki committed:
Merge pull request #10 from agileware-jp/3.4-feature/20747

ガントチャートのチケット編集フォームなどからチケットを編集したとき、チケットに変更履歴が残るようにした

maimai77 committed:
skip assignments when custom field columns are absent or ignored (#12)

maimai77 committed:
rubocop autocorrect (#13)

maimai77 committed:
rubocop autocorrect (#14)
